### PR TITLE
Add Genesis::Commands unit tests

### PR DESCRIPTION
**Pull request by wayneeseguin**
## Summary

- 36 subtests with 191 assertions covering command registration, option parsing, scope resolution, environment setup, help generation, and constants
- All `run_command` dependencies unit tested per FWT-570 acceptance criteria
- Discovered latent bug: `_u2d()` not exported from Genesis.pm but called in `Commands::get_options`

## Test plan

- [x] `prove -l t/unit-tests/genesis_commands-core.t` — all 36 subtests pass
- [x] `make unit-tests` — zero regressions (37 files, 545 tests)
- [x] Code review feedback addressed (not_ok convention, coderef dispatch, undef edge case)

Fixes: [FWT-570](https://fivetwenty.atlassian.net/browse/FWT-570)

### CODE REVIEW

File: t/unit-tests/genesis_commands-core.t [hunk 1]
```diff
@@ -0,0 +1,769 @@
...
+
+use lib 'lib';
+use lib 't';
+use helper;
+use Test::More;
+use Test::Deep;
+use Test::Exception;
+use Test::Output;
+use Test::Exit;
+use Cwd qw/getcwd abs_path/;
```
[the remark this answers is not in the document]
Comment (wayneeseguin) [reply t/unit-tests/genesis_commands-core.t]: Good catch — removed `getcwd` from the import. Fixed in 1142509.

File: t/unit-tests/genesis_commands-core.t [hunk 2]
```diff
@@ -0,0 +1,769 @@
...
+
+	# Sliced
+	my $slice = get_options('flag', 'name');
+	ok(exists $slice->{flag}, "flag in slice");
+	is($slice->{name}, 'hello', "name in slice");
+	ok(!exists $slice->{count}, "count excluded from slice");
+
+	# NOTE: get_options has an underscore-to-dash fallback via _u2d(),
+	# but _u2d is a private function in Genesis.pm not exported to Commands.
+	# This is a latent bug -- the fallback path would fail at runtime.
```
[the remark this answers is not in the document]
Comment (wayneeseguin) [reply t/unit-tests/genesis_commands-core.t]: Great suggestion. Adopted with one addition: `reset_commands_state()` before defining the test command to avoid state leaks from the preceding assertions in the same subtest.

The TODO test now surfaces the exact error (`Undefined subroutine &Genesis::Commands::_u2d`) without breaking the suite, documenting both the expected behavior and the missing export. Fixed in 1142509.
